build fail looks real
```
======================================================================
FAIL: test_simple_case_management (corehq.apps.userreports.tests.test_app_manager_integration.AppManagerDataSourceConfigTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/tests/test_app_manager_integration.py", line 65, in test_simple_case_management
    self.assertEqual(sample_doc[_get_column_property(result.column)], result.value)
AssertionError: 2 != u'2'
----------------------------------------------------------------------
```

--- a/corehq/apps/userreports/expressions/getters.py
+++ b/corehq/apps/userreports/expressions/getters.py
@@ -108,6 +108,8 @@ def transform_from_datatype(datatype):
     identity = lambda x: x
     return {
         'date': transform_date,
+        'datetime': transform_datetime,
+        'decimal': transform_decimal,
         'integer': transform_int,
         'string': unicode,
     }.get(datatype) or identity

--- a/corehq/apps/userreports/expressions/getters.py
+++ b/corehq/apps/userreports/expressions/getters.py
@@ -101,6 +101,18 @@ def transform_unicode(item):
         return None
 
 
+def transform_from_datatype(datatype):
+    """
+    Given a datatype, return a transform for that type.
+    """
+    identity = lambda x: x
+    return {
+        'date': transform_date,
+        'integer': transform_int,
+        'string': unicode,
+    }.get(datatype) or identity
+
+
 def getter_from_property_reference(spec):
     if spec.property_name:
         assert not spec.property_path, \

--- a/corehq/apps/userreports/expressions/specs.py
+++ b/corehq/apps/userreports/expressions/specs.py
@@ -24,7 +24,7 @@ class ConstantGetterSpec(JsonObject):
 class PropertyNameGetterSpec(JsonObject):
     type = TypeProperty('property_name')
     property_name = StringProperty(required=True)
-    datatype = DataTypeProperty(required=False, default="string")
+    datatype = DataTypeProperty(required=False)
 
     @property
     def expression(self):
@@ -39,7 +39,7 @@ class PropertyNameGetterSpec(JsonObject):
 class PropertyPathGetterSpec(JsonObject):
     type = TypeProperty('property_path')
     property_path = ListProperty(unicode, required=True)
-    datatype = DataTypeProperty(required=False, default="string")
+    datatype = DataTypeProperty(required=False)
 
     @property
     def expression(self):

--- a/corehq/apps/userreports/expressions/specs.py
+++ b/corehq/apps/userreports/expressions/specs.py
@@ -6,11 +6,8 @@ from corehq.apps.userreports.expressions.getters import (
     DictGetter,
     NestedDictGetter,
     TransformedGetter,
-)
-from corehq.apps.userreports.indicators.specs import (
-    DataTypeProperty,
-    _transform_from_datatype,
-)
+    transform_from_datatype)
+from corehq.apps.userreports.indicators.specs import DataTypeProperty
 from corehq.apps.userreports.specs import TypeProperty, EvaluationContext
 from corehq.util.quickcache import quickcache
 from dimagi.utils.couch.database import get_db
@@ -31,7 +28,7 @@ class PropertyNameGetterSpec(JsonObject):
 
     @property
     def expression(self):
-        transform = _transform_from_datatype(self.datatype)
+        transform = transform_from_datatype(self.datatype)
         getter = DictGetter(self.property_name)
         return TransformedGetter(getter, transform)
 
@@ -46,7 +43,7 @@ class PropertyPathGetterSpec(JsonObject):
 
     @property
     def expression(self):
-        transform = _transform_from_datatype(self.datatype)
+        transform = transform_from_datatype(self.datatype)
         getter = NestedDictGetter(self.property_path)
         return TransformedGetter(getter, transform)
 

--- a/corehq/apps/userreports/indicators/specs.py
+++ b/corehq/apps/userreports/indicators/specs.py
@@ -1,18 +1,14 @@
 from jsonobject import JsonObject, StringProperty, ListProperty, BooleanProperty, DictProperty
 from jsonobject.exceptions import BadValueError
-from corehq.apps.userreports.expressions.getters import (
-    getter_from_property_reference,
-    TransformedGetter,
-    transform_date,
-    transform_datetime,
-    transform_decimal,
-    transform_int,
-    transform_unicode,
-)
+from corehq.apps.userreports.expressions.getters import TransformedGetter, getter_from_property_reference, \
+    transform_from_datatype
 from corehq.apps.userreports.operators import IN_MULTISELECT, EQUAL
 from corehq.apps.userreports.specs import TypeProperty
 
+
 DATA_TYPE_CHOICES = ['date', 'datetime', 'string', 'integer', 'decimal']
+
+
 def DataTypeProperty(**kwargs):
     """
     Shortcut for valid data types.
@@ -68,7 +64,7 @@ class RawIndicatorSpec(PropertyReferenceIndicatorSpecBase):
 
     @property
     def getter(self):
-        transform = _transform_from_datatype(self.datatype)
+        transform = transform_from_datatype(self.datatype)
         getter = getter_from_property_reference(self)
         return TransformedGetter(getter, transform)
 
@@ -82,7 +78,7 @@ class ExpressionIndicatorSpec(IndicatorSpecBase):
 
     def parsed_expression(self, context):
         from corehq.apps.userreports.expressions.factory import ExpressionFactory
-        transform = _transform_from_datatype(self.datatype)
+        transform = transform_from_datatype(self.datatype)
         expression = ExpressionFactory.from_spec(self.expression, context)
         return TransformedGetter(expression, transform)
 
@@ -94,13 +90,3 @@ class ChoiceListIndicatorSpec(PropertyReferenceIndicatorSpecBase):
 
     def get_operator(self):
         return IN_MULTISELECT if self.select_style == 'multiple' else EQUAL
-
-
-def _transform_from_datatype(datatype):
-    return {
-        'date': transform_date,
-        'datetime': transform_datetime,
-        'string': transform_unicode,
-        'decimal': transform_decimal,
-        'integer': transform_int,
-    }.get(datatype)

